src/operators/semesterOperations.ts

```diff
@@ -0,0 +1,50 @@
+import Semester, { SemesterInterface } from '../models/semester';
+
+export const create = async (semester: SemesterInterface) => {
+	try {
+		await Semester.create(semester);
+	} catch (err) {
+		console.log(err);
+		throw 'ERROR: semester not created';
+	}
+};
+
+// NOTE: any reference to key is referring to the hash key, or the `id` parameter in all the models.
+export const read = async (key: string) => {
+	// add some kind of security check here?
+	const semester = await Semester.get(key);
+	if (semester) {
+		return semester;
+	} else {
+		throw 'ERROR: could not read semester';
+	}
+};
+
+export const update = async (data: Partial<SemesterInterface>) => {
+	let updated = false;
+	if (data.id) {
+		const key = data.id;
+		const semester = await Semester.get(key);
+		if (semester && (await Semester.update(data))) {
+			updated = true;
+		}
+	}
+	if (!updated) {
+		throw 'ERROR: semester not updated';
+	}
+};
```

Comment:
Could this be condensed to 
```ts
export const update = async (data: Partial<SemesterInterface>) => {
    let updated = false;
    if (data.id) {
        const key = data.id;
        const semester = await Semester.get(key);
        if (semester) {
            await Semester.update(data);
        }
    } else {
	throw 'ERROR: semester not updated';
    }
};
```
? Just curious if this breaks something!

Reply:
I think the difference between what I wrote and your suggestion is that I am catching both the case where id is not provided and also the case where the update call fails.